compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2294,7 +2294,7 @@ public void visit(BLangFail failNode) {
             // Check at module level.
             return;
         }
-        typeChecker.checkExpr(failNode.expr, env);
+        typeChecker.checkExpr(failNode.expr, env, symTable.noType, new Stack<>());
```

Comment:
I think it will be more clear if we can write a `checkExpr` method which has 2 params(expr and env) and call this function within new functions.